### PR TITLE
fix: regenerate csrf if flow has expired

### PR DESCRIPTION
**Pull request by splaunov**
When user clicks the verification link sent over email and the link has expired, then Kratos creates new verification flow and browser is redirected to verification ui with the new flow id set as a URL parameter.
The problem is that the verification UI cannot fetch the new flow as it has no csrf token set. So, browser gets 403 "Forbidden" error.

This PR eliminates csrf checking if verification flow has no csrf token.

## Checklist

- [ x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ x] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [ x] I have read the [security policy](../security/policy).
- [ x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](docs/docs).

### CODE REVIEW

**Review by aeneasr — CHANGES_REQUESTED** [review]

**Comment by aeneasr** [comment]
While the PR is being worked on I will mark it as a draft. That declutters our review backlog :)

Once you're done with your changes and would like someone to review them, mark the PR as ready and request a review from one of the maintainers.

Thank you!

File: selfservice/flow/verification/handler.go [hunk 1]
```diff
@@ -241,7 +241,7 @@ func (h *Handler) fetch(w http.ResponseWriter, r *http.Request, _ httprouter.Par
 	// Browser flows must include the CSRF token
 	//
 	// Resolves: https://github.com/ory/kratos/issues/1282
-	if req.Type == flow.TypeBrowser && !nosurf.VerifyToken(h.d.GenerateCSRFToken(r), req.CSRFToken) {
+	if req.Type == flow.TypeBrowser && req.CSRFToken != "" && !nosurf.VerifyToken(h.d.GenerateCSRFToken(r), req.CSRFToken) {
```
[the remark this answers is not in the document]
Comment (splaunov) [reply selfservice/flow/verification/handler.go]: Have extended test to cover this case and added token regeneration to the expired flow error handling logic.

**Review by aeneasr — DISMISSED** [review]

**Comment by aeneasr** [comment]
@jonas-jonas do we need to do this in the code strategy too?

**Comment by aeneasr** [comment]
@splaunov sorry for the late review, I am just catching ip on PRs! If the merge conflicts are addressed I will merge this immedia!

**Comment by jonas-jonas** [comment]
Yes, recovery should need that too: https://github.com/ory/kratos/blob/master/selfservice/flow/recovery/error.go#L82

I can take a look, or if you want to do it, you can do that in here. @splaunov :)

**Comment by splaunov** [comment]
The fix has been merged form some other branch.
Only the test change has been left not merged.

**Review by aeneasr — APPROVED** [review]

**Pull request merged**